DataFormats/Headers/include/Headers/NameHeader.h

```diff
@@ -57,7 +57,7 @@ struct NameHeader : public BaseHeader {
     *lastByte =
       reinterpret_cast<const uint8_t*>(this) + sizeof(NameHeader) - reinterpret_cast<const uint8_t*>(&name[N]);
     // here we actually want a null terminated string
-    strncpy(name, in.c_str(), N);
+    strncpy(name, in.c_str(), N < in.length() ? N : in.length());
```

Comment:
Is this actually right? Don't we zero pad?

Reply:
Thanks, but I think the padding is not a problem in this `NameHeader` example. But the string must be zero terminated, and that might be missing, so that would be a real bug.

Reply:
```suggestion
    strncpy(name, in.c_str(), N);
```

Reply:
Absolutely correct, the actual warning is also correct, it is pointing to where NameHeader is used and initialized with strings of the same or bigger length than the template parameter N. Thanks for spotting this.